torch/serialization.py

```diff
@@ -1010,10 +1010,10 @@ def load(
                     opened_file.seek(orig_position)
                     return torch.jit.load(opened_file, map_location=map_location)
                 if mmap:
-                    if not isinstance(f, str):
-                        raise ValueError("f must be a string filename in order to use mmap argument")
+                    if not _is_path(f):
```

Comment:
Why not use os.PathLIke here?

Reply:
Yeah, I used this function because it was simply already defined and used in this module. Can definitely replace it with instance check against PathLike if that's desired.

Reply:
Nah, this is fine but we may want to update the isinstance check at some point.

Reply:
Okay, so the other issue here is that mypy doesn't realize that ispath is narrowing the type, you may need to change the return type of ispath to be a 'typing_extension.TypeGuard[Union[...]' with proper annotations in the Union to tell mypy what types the input actually are if it returns True.

Reply:
Thanks @Skylion007
I added the type guard